sphinx/builders/__init__.py

```diff
@@ -420,8 +421,20 @@ def read(self) -> list[str]:
             self._read_serial(docnames)
 
         if self.config.root_doc not in self.env.all_docs:
-            raise SphinxError('root file %s not found' %
-                              self.env.doc2path(self.config.root_doc))
+            root_doc_path = self.env.doc2path(self.config.root_doc)
+            if any(fnmatch(root_doc_path, pat) or root_doc_path.endswith(pat)
```

Comment:
I don't think we want `root_doc_path.endswith(pat)`, otherwise if you have `root_doc_path = 'index.rst'` and one of your pattern is simply `rst`, then it doesn't match the pattern (the correct pattern would be `*.rst`). 

Instead, you should check that `root_doc_path != pat` (see https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-exclude_patterns for examples).

Reply:
I'm sorry I didn't check it before and I approved the PR.